aries-site/src/examples/templates/dashboards/content/chartCards/utils/hooks.js

```diff
@@ -0,0 +1,129 @@
+import { useEffect, useMemo, useState } from 'react';
+import {
+  convertDatesToFeatures,
+  defaultWindow,
+  linearRegression,
+  mean,
+  REPORT_WINDOW_MAP,
+} from '.';
+
+export const useCost = values => {
+  const [meanCost, setMeanCost] = useState(null);
+  const [lastMonthCost, setLastMonthCost] = useState(null);
+  const [projectedCost, setProjectedCost] = useState(null);
+
+  // Calculate cost stats
+  useEffect(() => {
+    const datapoints = [];
+    if (values) {
+      Object.keys(values).forEach(key => {
+        datapoints.push({ x: values[key].key, y: values[key].value });
+      });
+
+      setMeanCost(mean(datapoints.map(point => point.y)));
+      setLastMonthCost(datapoints[datapoints.length - 1].y);
+
+      const lastMonth = new Date(datapoints[datapoints.length - 1].x);
+      // Want last month +1, but also need to compensate for 0-based index,
+      // therefor +2.
+      const nextMonth = new Date(
+        lastMonth.getMonth() === 11
+          ? lastMonth.getFullYear() + 1
+          : lastMonth.getFullYear(),
+        (lastMonth.getMonth() + 2) % 12,
+        0,
+      );
+      /* Project next month's cost by fitting a regression line to our
+       * data. Dates are converted to numerical representations so that
+       * least-squares regression can interpret dates as the dependent
+       * variable. */
+      const dateMap = convertDatesToFeatures(
+        datapoints.map(point => point.x),
+        'month',
+        3,
+      );
+      /* y = mx + b, where y = projected cost, x = month, m = slope,
+       * and b = intercept. */
+      const { m, b } = linearRegression(
+        datapoints.map(point => ({
+          x: dateMap.get(point.x),
+          y: point.y,
+        })),
+      );
+      const costNextMonth =
+        m *
+          dateMap.get(
+            `${nextMonth.getMonth() +
+              1}/${nextMonth.getDate()}/${nextMonth.getFullYear()}`,
+          ) +
+        b;
+      setProjectedCost(costNextMonth.toFixed(2));
+    }
+  }, [values]);
+
+  return { meanCost, lastMonthCost, projectedCost };
+};
+
+export const useDataSet = (consumptionData, reportWindow) => {
+  const values = useMemo(() => {
+    if (consumptionData) {
+      const nextValues = [];
```

Comment:
This plus following suggestion:

```suggestion
    const nextValues = [];
    if (consumptionData) {
```